modules/lib/core/index.d.ts

```diff
@@ -1,3 +1,5 @@
+export type { ResourceKey } from '@enonic-types/global';
```

Comment:
Is it possible to make this dependency go the other way (so `ResourceKey` would still live in core)?

I think having core not depend on anything makes it really flexible. Now we would automatically be pulling in everything from global too.

Reply:
I was the other way around yesterday (aka global depended upon core) but we flipped it after a meeting.

> Now we would automatically be pulling in everything from global too.
So since global now is a global modifying module, simply importing from it, will add globals. Hmm.

Reply:
Yeah I don't like the side-effect of importing from core, and getting the global scope modified more than expected.

Reply:
By flipping it back so that global depends on core, then XpLayoutMap, XpPageMap, XpPartMap and XpXData will also exist when setting typeRoot to @enonic-types or including @enonic-types/global in types.

But that seems ok to me.

Reply:
Can anyone think of a reason why @enonic-types/global can't depend upon @enonic-types/core?

Reply:
I think as soon as global.d.ts does an import, it can no longer be an "global declaration file" but needs to be a "global modifying module", (which it currently is in this PR) (and which core is too).

ASFAIK "global modifying modules" aren't supposed to declare new globals, only augment existing ones, but it seems to work though.

Reply:
According to Gemini:

> Can TypeScript global modifying modules declare new global constants?
No

> Can TypeScript global modifying modules declare the shape of new global constants?
Yes

Since global.d.ts only declares the shape of `app` not the value, it's ok.

Reply:
> Can anyone think of a reason why @enonic-types/global can't depend upon @enonic-types/core?

You - as you said it would e breaking.

Reply:
It's "breaking" in the sense that a new dependency is added. In such that all @enonic-types packages must be of the same version.

It was also "breaking" because the build system tsconfig.json file "depended" upon the previous release of @enonic-types/core

Reply:
`ScriptValue` and `ResourceKey` technically belong in core - they represent Java classes exactly and there is always a Java object behind them.

`XpLayoutMap`, `XpPageMap`, `XpPartMap` and `XpXData` - they have similar classes in Java, but in JS they are `Map`, not actual Java objects - but core uses this types for communication in multiple methods

` XpBeans`, `XpLibraries` abstractions don't have functional value fore core - they are not used in core and can be left in global.


The more I look it this, the more I agree, that global needs core types, not core needs global types. But this is how it should be, I have no idea, if it is possible to do without breaking compatibility.

BTW ResourceKey.getApplication return type is `ApplicationKey`, `string`. 
`ApplicationKey` type has one method `getName()` - which is string

Reply:
I've flipped it back, and it seems to work.

Reply:
> ...will also exist when setting typeRoot to @enonic-types

I don't believe that there is any reason to set `compilerOptions.typeRoot` now, since we rely on `compilerOptions.paths` to be able to resolve all library types. (I also just use `compilerOptions.types` to add my global types).